share/spice/skyscanner_flight_search/skyscanner_flight_search.js

```diff
@@ -0,0 +1,155 @@
+            };
+        }
+        var settings = {
+            "async": true,
+            "crossDomain": true,
+            // "url": "https://gateway.skyscanner.net/travel-api/v1/entities?external_ids=" + listOfCountryIds + "&enhancers=images&apikey=09fd8de5844d4b1d982a320ad5dee5b8",
+            "method": "GET",
+            "headers": {
+                "cache-control": "no-cache",
+                "postman-token": "cf3cb5f2-92ff-15a0-3331-6b014e83abba"
+            }
+        };
+
+        //$.ajax(settings).success(function (response) {
+          
+        $.getJSON('/js/spice/skyscanner_images/' + listOfCountryIds, function(response) {
+            var flights = listOfRoutes.map(function(route,index) { return build_flight_route(route, index, response); });
+            
+            // Fail if no flights to show
+            if (flights.length === 0) {
+                return Spice.failed('No flights found');
+            }
+        
+            // Render the response
+            DDG.require('moment.js', function(){
+                Spice.add({
+                id: 'skyscanner_flight_search',
+                // This is the result tab header
+                name: 'Flights',
+                // data to pass to the template
+                data: flights,
+                meta: {
+                    //itemType: 'Results ' + l('Results'),
+                    sourceIcon: true,
+                    sourceName: 'Skyscanner',
+                    sourceUrl: 'http://skyscanner.net',
+                    total: flights,
+                    // You can either show Results: or customised text (primaryText)
+                    //itemType: (flights.length === 1) ? 'Result' : 'Results',
+                    primaryText: "Skyscanner's best deals from " + origin_country + "",
+                    // add regex to retrieve destination from search query
+                    searchTerm: 'flights to destination_city',
+                },
+
+                templates:{
+                    group: 'media',
+                    detail: false,
+                    item_detail: false,
+                    options: {
+                        description: false,
+                        footer: Spice.skyscanner_flight_search.footer,
+                    }
+                },
+
+                normalize: function(item) {
+                    return {
+                        disclaimer: "found " + moment(item.flight_price_age).fromNow() + " ago",
+                        title: item.flight_destination_city,
+                        image: item.flight_destination_city_image,
+                        url: "http://partners.api.skyscanner.net/apiservices/referral/v1.0/GB/GBP/en-GB/" + item.flight_origin_code + "/" + item.flight_destination_code + "/" 
+                            + moment(item.flight_outbound_date).format("YYYY-MM-DD") + "/" + moment(item.flight_return_date).format("YYYY-MM-DD") + "?apiKey=te1561648834359",
```

Comment:
> That's only a short version of the api key (16 chars instead of the full 32) to avoid exposing the full api key in the redirect url.

Oh, okay I hadn't noticed that. I'm more comfortable knowing it's not the full API key 👍 

> Would you prefer me to add it as {{ENV{DDG_SPICE_SKYSCANNER_SHORT_APIKEY}}}?

Unfortunately that approach won't work on the frontend, we don't have any way to hide API keys, but so far we haven't needed to.

@samheyman does this URL require our API key? It seems to work without that portion of the URL? Is this for SkyScanner to validate the queries are coming from a valid source?